profiles/views.py

```diff
@@ -61,28 +59,35 @@ class ProfileList(ListCreateAPIView):
      include_all: bool.
     """
     serializer_class = ProfileSerializer
-    permission_classes = (IsAuthenticatedOrReadOnly, )
+    permission_classes = [IsAuthenticatedOrReadOnly & SavedCompaniesListPermission]
     pagination_class = ForumPagination
 
     def get_queryset(self):
+        query_params_serializer = QueryParamSerializer(data=self.request.query_params)
         company_type = self.request.query_params.get("company_type")
         activity_type = self.request.query_params.get("activity_type")
+        userid = self.request.query_params.get("userid")
         HEADER_ACTIVITIES = ["producer", "importer", "retail", "horeca"]
 
-        if self.request.query_params:
+        queryset = Profile.objects.filter(is_deleted=False).order_by("profile_id")
+       
+        if userid:
             try:
-                userid = self.request.query_params.get('userid')
-                return Profile.objects.filter(person_id=userid)
+                return queryset.filter(person_id=userid)
             except ValueError:
-                raise ValidationError(detail='Bad request')
+                pass
```

Comment:
query set in this situation can't raise ValueError, this try-except is not needed here

Reply:
But if we pass as "userid" value that not an integer, ValueError occurs. So try/except lets us to avoid this error and to return queryset. Or there is another way to solve this issue?